compiler/modules/bir/bir.bal

```diff
@@ -92,6 +97,22 @@ public type FunctionSignature readonly & record {|
     SemType? restParamType = ();
 |};
 
+public type Region record {|
+    Label entry;
+    Label? exit = ();
+    RegionIndex? parent = ();
+    RegionKind kind;
+|};
+
+public enum RegionKind {
+    # Region that represents a basic loop.
```

Comment:
What do you mean by basic? do you mean loop only contains REGION_SIMPLE ?

Reply:
No did not mean that. 'basic' does not have a special meaning. I was copying the definition used by the Emscripten paper.

Reply:
Don't copy the Emscripten defn. Describe it in terms of BIR graph structure.

Reply:
Updated